### PR TITLE
fix: api profile inheritance

### PR DESCRIPTION
**Pull request by maximnaidenov**

### CODE REVIEW

File: src/uiveri5.js [hunk 1]
```diff
@@ -310,6 +312,41 @@ function run(config) {
...
+            return res.value;
+          });
+      };
+
+      browser.executeScriptLogResult = function executeScriptLogResult(scriptName,params) {
+        var code = clientsidescripts[scriptName];
+        params = params || {};
+        browser.controlFlow().execute(function () {
+          logger.debug('Execute script: ' + scriptName + ' with params: ' + JSON.stringify(params));
+          logger.trace('Execute script code: \n' + JSON.stringify(code));
```
Comment (tsaleksandrova) [inline src/uiveri5.js]: i'd rather log the name as it was before

File: conf/api.profile.conf.js [hunk 2]
```diff
@@ -1,13 +1,13 @@
...
 
-profile.matchers = profile.matchers.concat(
+baseProfile.matchers = [
   {name: './api/toHaveHttpBody'},
   {name: './api/toHaveHttpHeader'},
   {name: './api/body'}
-);
+];
 
-profile.plugins.push({name: '../src/api/requestPlugin'});
```
Comment (tsaleksandrova) [inline conf/api.profile.conf.js]: augment the plugins and matchers, don't replace them

File: src/uiveri5.js (hunk 1 above)
Comment (maximnaidenov) [reply src/uiveri5.js]: scriptName is logged in a similar way, we just avoid the duplication in the script name and refefence ?

File: conf/api.profile.conf.js (hunk 2 above)
Comment (maximnaidenov) [reply conf/api.profile.conf.js]: We would need more code to  handle the missing array case. I plan to handle it generically in an upcoming  profile inheritance feature, this is more like a fix to immediate issues in api profile.

File: src/control.js [hunk 3]
```diff
@@ -1,25 +1,19 @@
...
 Control.prototype.getProperty = function (property) {
-  return this.getWebElements().then(function (webElements) {
-    // at least one element is found, elsewise webdriver.error.ErrorCode.NO_SUCH_ELEMENT is thrown
-    return webElements[0].getAttribute('id');
-  }).then(function (elementId) {
-    return browser.executeScriptWithDescription(clientsidescripts.getControlProperty, 'Control.getProperty', {
-      elementId: elementId,
-      property: property
+  return this.getWebElements()
+    .then(function (webElements) {
```
[the remark this answers is not in the document]
Comment (maximnaidenov) [reply src/control.js]: I personally like it as it gives more visibility of the "sequencing" but will try to comply with used syntax.

File: conf/api.profile.conf.js (hunk 2 above)
Comment (tsaleksandrova) [reply conf/api.profile.conf.js]: wouldn't it be just a check for existence and array type?

Comment (maximnaidenov) [reply conf/api.profile.conf.js]: yes, it always starts with "just ...", we will do it correctly but not in the scope of this task.

**Pull request merged**